diff --git a/engine/utilities/intutils.h b/engine/utilities/intutils.h
index e6cf4986d..7e0fef578 100644
--- a/engine/utilities/intutils.h
+++ b/engine/utilities/intutils.h
@@ -225,34 +225,17 @@ struct IntOfSize<8> {
     typedef uint64_t utype;
 };
 
-#if defined(INTERNAL___INT128_FOUND)
-    #define INT128_AVAILABLE
-    template <>
-    struct IntOfSize<16> {
-        typedef __int128 type;
-        typedef __uint128 utype;
-    };
-#elif defined(INTERNAL___INT128_T_FOUND)
-    #define INT128_AVAILABLE
-    template <>
-    struct IntOfSize<16> {
-        typedef __int128_t type;
-        typedef __uint128_t utype;
-    };
-#elif defined(INTERNAL_INT128_T_FOUND)
-    #define INT128_AVAILABLE
-    template <>
-    struct IntOfSize<16> {
-        typedef int128_t type;
-        typedef uint128_t utype;
-    };
+#if ULONG_MAX > 4294967295
+template<>
+struct IntOfSize<16> {
+#if __GNUC__
+    __extension__ typedef __int128_t type;
+    __extension__ typedef __uint128_t utype;
 #else
-    #undef INT128_AVAILABLE
-    template <>
-    struct IntOfSize<16> {
-        typedef void type;
-        typedef void utype;
-    };
+    typedef __int128 type;
+    typedef __uint128 utype;
+#endif
+};
 #endif
 
 #endif // __DOXYGEN

diff --git a/python/regina/__init__.py b/python/regina/__init__.py
index 272c4b8d3..6e9c5dc9f 100644
--- a/python/regina/__init__.py
+++ b/python/regina/__init__.py
@@ -69,4 +69,15 @@ else:
 
+# In sageRegina, the census files are supplied differently.
+# Set the location here.
+try:
+    from .pyCensus import __path__ as _pyCensusPath
+    GlobalDirs.setDirs(
+        GlobalDirs.home(),
+        GlobalDirs.pythonModule(),
+        _pyCensusPath[0])
+except ImportError:
+    pass
+
 def reginaSetup(quiet = False, readline = True, banner = False,
                 snappyPath = True, namespace = None, builtinOpen = True):
     """
